### ISSUE TITLE
Create Instructions on how to setup a Trinity node for development

### ISSUE THREAD

**Issue by cburgdorf** [ethereum/py-evm#435]
### What is wrong?

The Trinity [Readme](https://github.com/ethereum/py-evm/blob/master/trinity/README.md) only lists a couple of environment variables but has no further information on how to setup and run a Trinity node.

This should be fixed to make it easier to onboard new contributors.

Here's what I tried to run the simple JSON-RPC server.

```
$ python3 trinity/rpc/server.py -db foo
Traceback (most recent call last):
  File "trinity/rpc/server.py", line 114, in <module>
    privkey=ecies.generate_privkey(),
  File "/home/cburgdorf/Documents/hacking/py-evm/evm/chains/base.py", line 292, in configure
    return super().configure(name, **overrides)
  File "/home/cburgdorf/Documents/hacking/py-evm/evm/utils/datatypes.py", line 18, in configure
    "not found on the base class `{2}`".format(cls.__name__, key, cls)
TypeError: The LightChain.configure cannot set attributes that are not already present on the base class. The attribute `privkey` was not found on the base class `<class 'p2p.lightchain.LightChain'>`
```

It's most likely related to the fact that I have not setup the environment variables. I actually have no idea how I would create a key to set as `TRINITY_NODEKEY` for instance.

The tinkerer in me tried to run `TRINITY_NODEKEY='foo' && python3 trinity/rpc/server.py -db foo` but that gave me the same error.

Long story short, the Readme should be newbie friendly and needs clear steps to get the node running.

### How can it be fixed

If someone can give me a couple of pointers I can happily compile that into a Readme update

**Comment by cburgdorf** [comment]
Ok, I just noticed that there are actually docs in the repo and they can be found online at http://py-evm.readthedocs.io.

That said, even the [Quickstart](http://py-evm.readthedocs.io/en/latest/quickstart.html) doesn't get me unstuck.

Basically leaves me at the same place as before:

```
$ python3 -i -m trinity.rpc.server -db /tmp/testnet.db
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/cburgdorf/Documents/hacking/py-evm/trinity/rpc/server.py", line 114, in <module>
    privkey=ecies.generate_privkey(),
  File "/home/cburgdorf/Documents/hacking/py-evm/evm/chains/base.py", line 292, in configure
    return super().configure(name, **overrides)
  File "/home/cburgdorf/Documents/hacking/py-evm/evm/utils/datatypes.py", line 18, in configure
    "not found on the base class `{2}`".format(cls.__name__, key, cls)
TypeError: The LightChain.configure cannot set attributes that are not already present on the base class. The attribute `privkey` was not found on the base class `<class 'p2p.lightchain.LightChain'>`
```

So, here's the deal, someone help me to resolve this and I make all needed changes to Readme and docs to decrease the likelihood of someone else struggling on this.

**Comment by gsalgado** [comment]
That's because `LightChain` no longer takes a `privkey` argument. That arg is now passed to `PeerPool`, which is then passed to `LightChain`. See `run_networking_process()` in trinity/main.py for an example

**Comment by cburgdorf** [comment]
Hey @gsalgado thank you for that hint. That means there probably needs something to be fixed in `trinity/rpc/server` then.

However, I made a bit of progress but there are lots of questions popping up in my head :)

1. I noticed I had to install `leveldb` as it isn't in the `requirements-dev.txt`. Can you confirm that this is intentionally not on the `requirements-dev.txt` because then I can compile it into some _prerequisites_ paragraph when I make updates to Readme/docs.

2. I understand that the `nodekey` identifies a node which is important for the p2p communication to work. Can you help me how to create one or point me to some docs?

3. The closest to running Trinity that I got so far goes like this

```
python3 -i -m trinity.main --ropsten --data-dir /tmp/trinity/data --trinity-root-dir /tmp/trinity
```
And then on the interactive console I type `main()` to kick things of. Is that the intended way to start the trinity process at this point?

4. I understand that `trinity/rpc/server.py` and `trinity/rpc/main.py` both implement different JSON-RPC APIs. What is the relationship to `trinity/main.py`? Is it intended that `trinity/main.py` will be the overall entrypoint of a trinity node and manages two spin up the two mentioned servers or is it intended to actually run something like `trinity/rpc/server.py` manually? At least with the current code base I don't see `trinity/main.py` managing these two server APIs and I see that `trinity/rpc/server.py` is intended to be called directly. 

I feel bad for asking all these questions but I promise to pay back ;)

**Comment by cburgdorf** [comment]
Poked around and ended up with another PR https://github.com/ethereum/py-evm/pull/437/files

**Comment by cburgdorf** [comment]
I'm waiting for the dust to settle in https://github.com/ethereum/py-evm/pull/382 and then I will update the docs / Readme to make sure everyone has a smooth experience on how to run trinity with the different options.

**Comment by cburgdorf** [comment]
Closing as this is now covered by http://py-evm.readthedocs.io/en/latest/contributing.html
